### PR TITLE
Support a reader tuple as an argument in to_rows/to_columns

### PR DESCRIPTION
**Pull request by jonatanklosko**
:cat2:

### CODE REVIEW

**Comment by jonatanklosko** [comment]
@josevalim with reader as an argument for `to_rows`/`to_columns`, we can do this:

```elixir
reader = Table.Reader.init(tabular)
rows = Table.to_rows(reader)
{_, meta, _} = reader
```

This makes me wonder if we should deprecate `_with_info` altogether, wdyt?

**Comment by josevalim** [comment]
> This makes me wonder if we should deprecate _with_info altogether, wdyt?

Good call, sounds good to me!

**Review by josevalim — APPROVED** [review]

File: lib/table.ex [hunk 1]
```diff
@@ -73,30 +73,25 @@ defmodule Table do
...
-      iex> info
-      %{columns: [:id, :name]}
+    tabular
+    |> init_reader!()
+    |> read_rows(only)
+  end
 
-  """
-  @spec to_rows_with_info(Reader.t(), keyword()) :: {Enumerable.t(), table_info()}
+  # TODO: remove in v0.2
```
Comment (jonatanklosko) [inline lib/table.ex:85]: I think the next version can be v0.1.2, so we remove on v0.2?

Comment (josevalim) [reply lib/table.ex:85]: Sounds good!

**Pull request merged**